kfs-core/src/main/java/edu/arizona/kfs/fp/service/impl/PaymentMethodGeneralLedgerPendingEntryServiceImpl.java

```diff
@@ -0,0 +1,455 @@
+            if ( objectCode == null ) {
+                LOG.fatal("Specified offset object code: " + chargeEntry.getChartOfAccountsCode() + "-" + chargeEntry.getFinancialObjectCode() + " does not exist - failed to generate foreign draft fee entries", new RuntimeException() );
+                return false;
+            }       
+            chargeEntry.setFinancialObjectTypeCode(objectCode.getFinancialObjectTypeCode());
+            
+            // Set the amount from the parameter
+            chargeEntry.setTransactionLedgerEntryAmount(feeAmount);
+            chargeEntry.setTransactionDebitCreditCode(reverseEntries?GL_CREDIT_CODE:GL_DEBIT_CODE);
+    
+            document.addPendingEntry(chargeEntry);
+            sequenceHelper.increment();
+    
+            // handle the offset entry
+            GeneralLedgerPendingEntry offsetEntry = new GeneralLedgerPendingEntry(chargeEntry);
+            getGeneralLedgerPendingEntryService().populateOffsetGeneralLedgerPendingEntry(document.getPostingYear(), chargeEntry, sequenceHelper, offsetEntry);
+    
+            document.addPendingEntry(offsetEntry);
+            sequenceHelper.increment();
+            
+            // Now, create the income entry in the AP Foreign draft fee account
+            
+            GeneralLedgerPendingEntry feeIncomeEntry = new GeneralLedgerPendingEntry(document.getGeneralLedgerPendingEntry(0));
+            feeIncomeEntry.setTransactionLedgerEntrySequenceNumber(sequenceHelper.getSequenceCounter());
+    
+            feeIncomeEntry.setChartOfAccountsCode(feeIncomeChartCode);
+            feeIncomeEntry.setAccountNumber(feeIncomeAccountNumber);
+            feeIncomeEntry.setFinancialObjectCode(feeIncomeObjectCode);
+            feeIncomeEntry.setFinancialSubObjectCode(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankFinancialSubObjectCode());
+            feeIncomeEntry.setSubAccountNumber(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankSubAccountNumber());
+            feeIncomeEntry.setProjectCode(GENERAL_LEDGER_PENDING_ENTRY_CODE.getBlankProjectCode());
+    
+            // retrieve object type
+            objectCode = getObjectCodeService().getByPrimaryIdForCurrentYear(feeIncomeChartCode, feeIncomeObjectCode);
+            if ( objectCode == null ) {
+                LOG.fatal("Specified income object code: " + feeIncomeChartCode + "-" + feeIncomeObjectCode + " does not exist - failed to generate foreign draft income entries", new RuntimeException() );
+                return false;
+            }
+            feeIncomeEntry.setFinancialObjectTypeCode(objectCode.getFinancialObjectTypeCode());       
+            feeIncomeEntry.setTransactionLedgerEntryAmount(feeAmount);
+            feeIncomeEntry.setTransactionDebitCreditCode(reverseEntries?GL_DEBIT_CODE:GL_CREDIT_CODE);
+            feeIncomeEntry.setFinancialBalanceTypeCode(KFSConstants.BALANCE_TYPE_ACTUAL);
```

Comment:
This constant has been deprecated. Should be getting this from SystemOptions. See LaborBalanceSummaryReportServiceImpl.getActualBalanceTypes(Integer fiscalYear).